Makefile

```diff
@@ -19,62 +19,63 @@ clean:
 	find . -name "__pycache__" -exec rm -rf {} +
 	find . -type d -empty -delete
 
-cov:
-	$(PYTHON) -m coverage report
+cov: dev
+	$(UV) run coverage report
 
-cov-html:
-	$(PYTHON) -m coverage html
+cov-html: dev
+	$(UV) run coverage html
 	open htmlcov/index.html
 
-cov-xml:
-	$(PYTHON) -m coverage xml
-
-deps: ensure-uv
-	$(UV) pip install --upgrade pip setuptools wheel -r requirements.txt -r requirements-dev.txt
+cov-xml: dev
+	$(UV) run coverage xml
 
 dev: ensure-uv
 	$(UV) pip install -e .
 
-docs:
+docs: ensure-uv
 	$(MAKE) -C ./docs
 
-ensure-uv:
+$(VIRTUAL_ENV):
+	$(UV) venv $(VIRTUAL_ENV)
+_ensure-uv-venv: _ensure-uv-cmd $(VIRTUAL_ENV)
+_ensure-uv-cmd:
 	@if ! command -v $(UV) >/dev/null; then \
-		$(PYTHON) -m ensurepip && $(PYTHON) -m pip install uv; \
+		$(PYTHON) -m ensurepip && $(PYTHON) -m pip install "uv >= 0.4.27"; \
 	fi
+ensure-uv: _ensure-uv-cmd _ensure-uv-venv
 
-fmt:
-	$(PYTHON) -m ruff check --fix
-	$(PYTHON) -m ruff format
+fmt: dev
```

Comment:
I have been hesitant to create these makefile target dependencies in the past due to the execution time overhead of running `dev` everytime. But it appears that the overhead is trival using this setup. Just to double check myself, can you think of any other reasons why we wouldn't want to run `dev` before each of these commands?

Reply:
There is a hiccup in running it. But I believe the advantages outweigh the small slowdown for each run